src/allmydata/test/test_storage_web.py

```diff
@@ -398,25 +415,25 @@ def _after_first_bucket(ignored):
         d.addCallback(lambda ign: renderDeferred(webstatus))
         def _check_html_in_cycle(html):
             s = remove_tags(html)
-            self.failUnlessIn("So far, this cycle has examined "
-                              "1 shares in 1 buckets (0 mutable / 1 immutable) ", s)
-            self.failUnlessIn("and has recovered: "
-                              "0 shares, 0 buckets (0 mutable / 0 immutable), "
-                              "0 B (0 B / 0 B)", s)
-            self.failUnlessIn("If expiration were enabled, "
-                              "we would have recovered: "
-                              "0 shares, 0 buckets (0 mutable / 0 immutable),"
-                              " 0 B (0 B / 0 B) by now", s)
-            self.failUnlessIn("and the remainder of this cycle "
-                              "would probably recover: "
-                              "0 shares, 0 buckets (0 mutable / 0 immutable),"
-                              " 0 B (0 B / 0 B)", s)
-            self.failUnlessIn("and the whole cycle would probably recover: "
-                              "0 shares, 0 buckets (0 mutable / 0 immutable),"
-                              " 0 B (0 B / 0 B)", s)
-            self.failUnlessIn("if we were strictly using each lease's default "
-                              "31-day lease lifetime", s)
-            self.failUnlessIn("this cycle would be expected to recover: ", s)
+            self.failUnlessIn(b"So far, this cycle has examined "
```

Comment:
`b"fdfds"` has a very specific meaning in Python 2 and 3: it means "this is bytes". Unlike Python 2, where bytes and unicode can be shoved together indiscriminately, you can't for example concatenate a byte string and unicode string in Python 3. So it's very important to make sure things that are supposed to be bytes are marked as bytes, and vice versa. And this has  nothing to do with ASCII or not-ASCII in Python 3, it's about the in-memory Python type.

So yes, it is necessary, and it will break if it's not marked as bytes.